server/tree.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+
+function isFile(content) {
+  return new Promise((resolve, reject) => {
+    fs.stat(content, (err, stats) => {
+      if (err) {
+        reject(err);
+      }
+
+      resolve(stats.isFile());
+    });
+  });
+}
+
+function formResponse(directoryContent, directoryPath ) {
+  return Promise.all(directoryContent.map(entry => {
+    return isFile(path.resolve(directoryPath , entry))
+      .then(ifFile => {
+        return {
+          type: ifFile ? 'file' : 'directory',
+          name: entry
+        };
+      });
+  }));
+}
+
+function readDirectory(directoryPath ) {
+  return new Promise((resolve, reject) => {
+    fs.readdir(directoryPath , (err, files) => {
+      if (err) {
+        reject(err);
+      }
+
+      formResponse(files, directoryPath ).then(resolve);
+    });
+  });
+}
+
+function getDirectoryContent(directoryPath ) {
+  return readDirectory(directoryPath);
```

Comment:
sorry, might have confused you in previous comment. I was suggesting splitting fs/io from forming result
so it would become something along following lines
```
function getDirectoryContent(directoryPath) {
   return readDirectory(directoryPath).then(files => formResponse(files, directoryPath))
}
```